### PR TITLE
[Placement group] Update the old placement group API usage to the new scheduling_strategy based API

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

Documentation should use the new API, not the old one that will be deprecated

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by scv119 — APPROVED** [review]

File: doc/source/ray-core/placement-group.rst [hunk 1]
```diff
@@ -215,7 +216,7 @@ Let's create a placement group. Recall that each bundle is a collection of resou
         f.remote()
 
         # Will be scheduled because 2 cpus are reserved by the placement group.
-        f.options(placement_group=pg).remote()
+        f.options(scheduling_strategy=PlacementGroupSchedulingStrategy(placement_group=pg)).remote()
```
Comment (scv119) [inline doc/source/ray-core/placement-group.rst:219]: consider break it into two lines?

File: doc/source/ray-core/placement-group.rst [hunk 2]
```diff
@@ -505,20 +510,22 @@ because they are scheduled on a placement group with the STRICT_PACK strategy.
...
         def parent():
             # The child task is scheduled with the same placement group as its parent
-            # although child.options(placement_group=pg).remote() wasn't called.
+            # although child.options(
+              scheduling_strategy=PlacementGroupSchedulingStrategy(placement_group=pg)
+            ).remote() wasn't called.
             ray.get(child.remote())
 
-        ray.get(parent.options(placement_group=pg).remote())
+        ray.get(parent.options(scheduling_strategy=PlacementGroupSchedulingStrategy(placement_group=pg)).remote())
```
Comment (scv119) [inline doc/source/ray-core/placement-group.rst:518]: same here

File: doc/source/ray-core/placement-group.rst [hunk 3]
```diff
@@ -505,20 +510,22 @@ because they are scheduled on a placement group with the STRICT_PACK strategy.
...
+      To avoid it, you should specify `options(scheduling_strategy=PlacementGroupSchedulingStrategy(placement_group=None))` in a child task/actor remote call.
 
       .. code-block:: python
 
         @ray.remote
         def parent():
             # In this case, the child task won't be
             # scheduled with the parent's placement group.
-            ray.get(child.options(placement_group=None).remote())
+            ray.get(child.options(scheduling_strategy=PlacementGroupSchedulingStrategy(placement_group=None)).remote())
```
Comment (scv119) [inline doc/source/ray-core/placement-group.rst:528]: ditto

**Comment by rkooo567** [comment]
Hmm @jjyao It seems like PlacementGroupStrategy class itself is not documented. Is this correct? Or do you know where I can find the API reference? (If not, I think I should update that API reference here too).

**Comment by jjyao** [comment]
`PlacementGroupStrategy` has class level documentation if that's what you are asking.

File: doc/source/ray-core/tasks/scheduling.rst [hunk 4]
```diff
@@ -16,7 +16,7 @@ If all nodes are infeasible, the task cannot be scheduled until feasible nodes a
 
 Placement Group
 ---------------
-If ``placement_group`` option is set then the task will be scheduled where the placement group is located.
+If ``scheduling_strategy=PlacementGroupSchedulingStrategy`` option is set then the task will be scheduled where the placement group is located.
```
Comment (jjyao) [inline doc/source/ray-core/tasks/scheduling.rst:19]: Could you also update actors/scheduling.rst?

File: doc/source/ray-core/placement-group.rst [hunk 5]
```diff
@@ -487,38 +516,66 @@ because they are scheduled on a placement group with the STRICT_PACK strategy.
...
+  set ``placement_group_capture_child_tasks`` to True.
 
   .. tabbed:: Python
 
       .. code-block:: python
 
-        # Create a placement group with the STRICT_SPREAD strategy.
-        pg = placement_group([{"CPU": 2}, {"CPU": 2}], strategy="STRICT_SPREAD")
+        import ray
+        import ray
```
Comment (jjyao) [inline doc/source/ray-core/placement-group.rst:528]: ?

Seems this code snippet is running by itself, can we use `.. literalinclude::` so it's testable.

**Comment by rkooo567** [comment]
@jjyao where is it documented though? I failed to find the documentation from the doc. (I searched the class, and there was no result). I am asking it because I'd like to add a link to the API reference.

**Review by jjyao — APPROVED** [review]
Let's fix `PlacementGroupSchedulingStrategy docstring`

**Pull request merged**